src/org/opensha/commons/calc/magScalingRelations/magScalingRelImpl/TMG2017_MagAreaRel.java

```diff
@@ -0,0 +1,304 @@
+      if (regime.equalsIgnoreCase("crustal"))
+    	if (( rake <= 45 && rake >= -45 ) || rake >= 135 || rake <= -135)
+    	  // strike slip
+    	  return  0.184;
+    	else if (rake > 0)
+    	  // thrust/reverse
+    	  return  0.121;
+    	else
+    	  // normal
+    	  return  0.181;
+      else
+    	// else subduction-interface
+      	return 0.150;
+    }
+   
+    public double getMagStdDev(Double rake,String regime) {
+    	this.setRegime(regime);
+    	this.setRake(rake);
+    	return getMagStdDev(); 
+    }
+
+      
+    /** 
+     * overload to additionally consider regime 
+     */
+    public double getMedianMag(double area,  double rake, String regime){	 
+    	this.setRegime(regime);
+    	this.setRake(rake);
+    	return this.getMedianMag(area);
+      }
+      
+    
+    
+    /**
+     * Computes the median rupture area from magnitude (for the previously set
+     * rake and regime values).
+     * @param mag - moment magnitude
+     * @return median area in km
+     */
+    public double getMedianArea(double mag){
+        if  (Double.isNaN(rake))
+           // this estimate is not available 
+           return Double.NaN;
+        if (regime.equalsIgnoreCase("crustal"))
+        	if (( rake <= 45 && rake >= -45 ) || rake >= 135 || rake <= -135)
+        		// strike slip
+        		return  Math.pow(10.0, -3.486 + 0.942*mag);
+        	else if (rake > 0)
+        		// shallow thrust/reverse
+        		return  Math.pow(10.0, -4.362 + 1.049*mag);
+        	else
+        		// normal
+        		return  Math.pow(10.0, -2.551 + 0.808*mag);
+        else
+        	//subduction-interface
+        	return Math.pow(10.0, -3.292 + 0.949*mag);
+
+    }
+
+   
+    /**
+     * Computes the standard deviation of log(area) (base-10) from magnitude
+     *  (for the previously set rake and regime values)
+     * @return standard deviation
+     */
+    public double getAreaStdDev() {
+    	return getMagStdDev(); 
+
+    }
+
+    public double getAreaStdDev(Double rake,String regime) {
+    	this.setRegime(regime);
+    	this.setRake(rake);
+    	return getMagStdDev(); 
+    }
+
+    
+    /** 
+     * overload to additionally consider regime 
+     */
+    public double getMedianArea(double mag,  double rake, String regime){	 
+    	this.setRegime(regime);
+    	this.setRake(rake);
+    	return this.getMedianArea(mag);
+      }
+      
+    
+    /**
+     * Returns the name of the object
+     *
+     */
+    public String getName() {
+    	String type;
+        if (Double.isNaN(rake))
+            type = "not available";
+        
+        if (regime.equalsIgnoreCase("crustal"))
+          if (( rake <= 45 && rake >= -45 ) || rake >= 135 || rake <= -135)
+            type = "strike-slip";
+          else if (rake > 0)
+            type = "shallow reverse-faulting";
+          else
+            type =  "normal-faulting";
+        else
+        	type = "interface";
+    	
+      return NAME +" for "+type+" events";
+    }
+    
+/*
```

Comment:
Nice! Would you like me to turn these into unit tests?

Reply:
Sure.  The test data would be:

Area,  R_mag,  N_Mag,  SS_Mag, Int_mag 
1,  4.158246,   3.157178,   3.700637,  3.468915 
500,  6.731144,   6.497488,   6.565786,  6.312929 
10000,  7.971401,   8.107673,   7.946921,  7.683878 
100000,  8.924690,   9.345297,   9.008493,  8.737619 
Mag,  R_Area,  N_Area,  SS_Area, Int_Area 
4.000000,  0.682339,   4.797334,   1.914256,  3.191538 
6.000000,  85.506671,   198.152703,   146.554784,  252.348077 
8.000000,  10715.193052,   8184.647881,   11220.184543,  19952.623150 
9.000000,  119949.930315,   52601.726639,   98174.794302,  177418.948089 

I think test for StdDev is not required.